blazor/datagrid/custom-binding.md

```diff
@@ -988,3 +988,418 @@ The following sample code demonstrates sending additional parameters to the cust
 
 ![Passing Additional Parameters to Custom Adaptor in Blazor DataGrid](./images/blazor-datagrid-passing-additional-parameter-to-custom-adaptor.gif)
+
+## DataTable Binding
+
+DataTable represents a table with data rows and columns.
+Data binding of the DataTable can be acheieved in the DataGrid component by providing the **CustomAdaptor** class and overriding the Read or ReadAsync method of the DataAdaptor abstract class.
+
+To bind DataTable to the datagrid, you can assign `TValue` to the **ExpandoObject**. 
+
+Refer to the following code example for how to bind the DataTable using custom adaptor.
+
+```cshtml
+@using Syncfusion.Blazor;
+@using Syncfusion.Blazor.Data
+@using Syncfusion.Blazor.Grids
+@using System.Dynamic;
+@using System.Data;
+
+<SfGrid TValue="ExpandoObject" ID="Grid" AllowSorting="true" AllowPaging="true" AllowFiltering="true" Toolbar="@(new List<string>() { "Search" })">
+    <SfDataManager AdaptorInstance="@typeof(CustomAdaptor)" Adaptor="Adaptors.CustomAdaptor"></SfDataManager>
+    <GridPageSettings PageSize="8"></GridPageSettings>
+    <GridColumns>
+        <GridColumn Field="OrderID" HeaderText="Order ID" IsPrimaryKey="true" ValidationRules="@(new ValidationRules{ Required=true, Number=true})" Width="120"></GridColumn>
+        <GridColumn Field="CustomerID" HeaderText="Customer Name" ValidationRules="@(new ValidationRules { Required=true})" Width="150"></GridColumn>
+        <GridColumn Field="OrderDate" HeaderText="Order Date" Format="d" Type="ColumnType.Date" Width="130"></GridColumn>
+        <GridColumn Field="EmployeeID" HeaderText="Employee ID" Width="120"></GridColumn>
+    </GridColumns>
+</SfGrid>
+
+@code {
+    public static DataTable dataTable { get; set; }
+    public static IQueryable DataSource;
+
+    protected override void OnInitialized()
+    {
+        dataTable = GetData();
+        DataSource = ToQueryableCollection(dataTable);      // Convert DataTable to IQueryable ExpandoObject list
+    }
+
+    // Implementing custom adaptor by extending the DataAdaptor class
+    public class CustomAdaptor : DataAdaptor
+    {
+        // Performs data Read operation
+        public override object Read(DataManagerRequest dm, string key = null)
+        {
+            // Performs Searching, Sorting, Filtering
+            DataSource = PerformDataOperation(dataTable, dm);
+
+            int count = DataSource.Cast<ExpandoObject>().Count();
+            if (dm.Skip != 0)
+            {
+                //Paging
+                DataSource = QueryableOperation.PerformSkip<object>((IQueryable<object>)DataSource, dm.Skip);
+            }
+            if (dm.Take != 0)
+            {
+                DataSource = QueryableOperation.PerformTake<object>((IQueryable<object>)DataSource, dm.Take);
+            }
+
+            return dm.RequiresCounts ? new DataResult() { Result = DataSource, Count = count } : (object)DataSource;
+        }
+    }
+
+    public static IQueryable ToQueryableCollection(DataTable dt)
```

Comment:
Before this code example, we need to explain  PerformDataOperation and then add the ToQueryableCollection method